schedule_create_transaction_e2e_test.go

```diff
@@ -106,630 +107,919 @@ func TestIntegrationScheduleCreateTransactionCanExecute(t *testing.T) {
+}
+
+func TestIntegrationScheduleCreateTransactionWaitForExpiry(t *testing.T) {
+	env := NewIntegrationTestEnv(t)
+	defer CloseIntegrationTestEnv(env, nil)
+
+	key, err := GeneratePrivateKey()
+	require.NoError(t, err)
+
+	createResponse, err := NewAccountCreateTransaction().
+		SetKey(key.PublicKey()).
+		SetInitialBalance(NewHbar(2)).
+		Execute(env.Client)
+	require.NoError(t, err)
+
+	transactionReceipt, err := createResponse.GetReceipt(env.Client)
+	require.NoError(t, err)
+
+	accountId := *transactionReceipt.AccountID
+
+	// Create the transaction
+	transaction := NewTransferTransaction().
+		AddHbarTransfer(accountId, NewHbar(1).Negated()).
+		AddHbarTransfer(env.Client.GetOperatorAccountID(), NewHbar(1)).
+		SetMaxTransactionFee(NewHbar(10))
+
+	scheduled, err := transaction.Schedule()
+	require.NoError(t, err)
+
+	// Schedule the transaction
+	resp, err := scheduled.
+		SetExpirationTime(time.Now().Add(10 * time.Second)).
+		SetWaitForExpiry(true).
+		Execute(env.Client)
+	require.NoError(t, err)
+
+	receipt, err := resp.SetValidateStatus(true).GetReceipt(env.Client)
+	require.NoError(t, err)
+
+	scheduleId := *receipt.ScheduleID
+	txId := *receipt.TransactionID
+
+	// Verify the transaction is not executed
+	scheduleInfo, err := NewScheduleInfoQuery().
+		SetScheduleID(scheduleId).
+		Execute(env.Client)
+	require.NoError(t, err)
+	require.Nil(t, scheduleInfo.ExecutedAt)
+
+	// Schedule sign
+	frozenSign, err := NewScheduleSignTransaction().
+		SetScheduleID(scheduleId).
+		FreezeWith(env.Client)
+	require.NoError(t, err)
+
+	resp, err = frozenSign.Sign(key).Execute(env.Client)
+	require.NoError(t, err)
+	_, err = resp.SetValidateStatus(true).GetReceipt(env.Client)
+	require.NoError(t, err)
+
+	// Verify the transaction is still not executed
+	scheduleInfo, err = NewScheduleInfoQuery().
+		SetScheduleID(scheduleId).
+		Execute(env.Client)
+	require.NoError(t, err)
+	require.Nil(t, scheduleInfo.ExecutedAt)
+
+	accountBalanceBefore, err := NewAccountBalanceQuery().
+		SetAccountID(accountId).
+		Execute(env.Client)
+	require.NoError(t, err)
+
+	time.Sleep(10 * time.Second)
+
+	accountBalanceAfter, err := NewAccountBalanceQuery().
+		SetAccountID(accountId).
+		Execute(env.Client)
+	require.NoError(t, err)
+	assert.Equal(t, accountBalanceBefore.Hbars.AsTinybar(), accountBalanceAfter.Hbars.AsTinybar()+100000000)
+
+	_, err = NewTransactionRecordQuery().
+		SetTransactionID(txId).
+		Execute(env.Client)
+	require.NoError(t, err)
+}
+
+func TestIntegrationScheduleCreateTransactionUpdateSignRequirements(t *testing.T) {
+	env := NewIntegrationTestEnv(t)
+	defer CloseIntegrationTestEnv(env, nil)
+
+	key1, err := GeneratePrivateKey()
+	require.NoError(t, err)
+
+	key2, err := GeneratePrivateKey()
+	require.NoError(t, err)
+
+	key3, err := GeneratePrivateKey()
+	require.NoError(t, err)
+
+	key4, err := GeneratePrivateKey()
+	require.NoError(t, err)
```

Comment:
nit: Different convention from the other test where Key list data is populated via loop. 
note: For future and past tests where we need to generate an array of keys maybe we can have something like `generateKeys` for populating KeyLists.